fs/rails-base-graphql-api
# Update login flow

### Summary

Added signin mutation that returns token and me fields on success. This endpoint will be used for authenticate a user.

### Test plan

List of steps to manually test introduced functionality:

* Open [GraphiQL App](https://www.electronjs.org/apps/graphiql)
* Set http://localhost:5000/graphiql as GraphQL endpoint
* Create user using signup mutation
* Make request using schema:
```
  mutation {
    signin(email: "user@example.com", password: "123456") {
      token
      me {
        id
        email
      }
    }
  }
```
* User token along with the user object should return

### Review notes

While reviewing pull-request (especially when it's your pull-request),
please make sure that:

- you understand what problem is solved by PR and how is it solved
- new tests are in place, no redundant tests
- DB schema changes reflect new migrations
- newly introduced DB fields have indexes and constraints
- there are no missed files (migrations, view templates)
- required ENV variables added and described in `.env.example` and added to Heroku
- associated Heroku review app works correctly with introduced changes

### Deploy notes

N/A

### References

N/A
